plugin/src/main/java/org/opensearch/ml/utils/SecurityUtils.java

```diff
@@ -0,0 +1,122 @@
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.collect.ImmutableList;
+
+@Log4j2
+public class SecurityUtils {
+
+    public static void validateModelGroupAccess(User user, String modelGroupId, Client client, ActionListener<Boolean> listener) {
+        if (modelGroupId == null || isAdmin(user)) {
+            listener.onResponse(Boolean.TRUE);
+            return;
+        }
+
+        List<String> userBackendRoles = user.getBackendRoles();
+
+        GetRequest getModelGroupRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
+        try {
+            client.get(getModelGroupRequest, ActionListener.wrap(r -> {
+                if (r != null && r.isExists()) {
+                    List<String> modelGroupBackendRoles = new ArrayList<>();
+                    String access = null;
+                    Map<String, Object> owner = null;
+                    Map<String, Object> source = r.getSourceAsMap();
+                    if (source.containsKey(MLModelGroup.ACCESS)) {
+                        access = (String) source.get(MLModelGroup.ACCESS);
+                    }
+                    if (source.containsKey(MLModelGroup.BACKEND_ROLES_FIELD) && source.get(MLModelGroup.BACKEND_ROLES_FIELD) != null) {
+                        modelGroupBackendRoles = (List<String>) source.get(MLModelGroup.BACKEND_ROLES_FIELD);
+                    }
+                    if (source.containsKey(MLModelGroup.OWNER)) {
+                        owner = (Map) source.get(MLModelGroup.OWNER);
+                    }
+                    if (access != null && StringUtils.equals(access, MLModelGroup.PUBLIC)) {
+                        listener.onResponse(Boolean.TRUE);
+                    } else if (access != null && StringUtils.equals(access, MLModelGroup.PRIVATE)) {
+                        if (isOwner(owner, user)) {
+                            listener.onResponse(Boolean.TRUE);
+                            return;
+                        }
+                        listener.onResponse(Boolean.FALSE);
+                    } else {
+                        listener
+                            .onResponse(
+                                userBackendRoles.stream().anyMatch(modelGroupBackendRoles.stream().collect(Collectors.toSet())::contains)
+                            );
+                    }
+                }
+            }, e -> {
+                log.error("Failed to validate Access", e);
+                listener.onFailure(new MLValidationException("Failed to validate Access"));
+            }));
+        } catch (Exception e) {
+            log.error("Failed to validate Access", e);
+            listener.onFailure(e);
+        }
+    }
+
+    public static boolean isAdmin(User user) {
+        if (user == null) {
+            return false;
+        }
+        if (CollectionUtils.isEmpty(user.getRoles())) {
+            return false;
+        }
+        return user.getRoles().contains("all_access");
```

Comment:
is it possible to have `ALL_ACCESS`

Reply:
User will all access is the admin and that is what we would check in this method